examples/sparse/hgnn.py

```diff
@@ -0,0 +1,95 @@
+"""
+Hypergraph Neural Networks (https://arxiv.org/pdf/1809.09401.pdf)
+"""
+import dgl
+import dgl.data
+import dgl.mock_sparse
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torchmetrics.functional import accuracy
+import tqdm
+
+dataset = dgl.data.CoraGraphDataset()
+
+graph = dataset[0]
+# The paper created the hypergraph by "each time one vertex in the
+# graph is selected as the centroid and its connected vertices
+# are used to generate one hyperedge including the centroid
+# itself".  In this case, the incidence matrix of the hypergraph
```

Comment:
An attempt to improve the description from L16-19

```suggestion
The paper created a hypergraph from the original graph. For each node in the original graph, a hyperedge in the hypergraph is created to connect this center node and all neighbors of the center node including itself. In this case, ...
```